lib/module.js

```diff
@@ -159,7 +161,7 @@ Module._findPath = function(request, paths, isMain) {
     if (!trailingSlash) {
       const rc = stat(basePath);
       if (rc === 0) {  // File.
-        if (!isMain) {
+        if (preserveSymlinks && !isMain) {
```

Comment:
Initially this change (as was mine #6536) was about reverting #5950. Now this commit just builds on #5950 which was wrong. 

In this diff one is under impression that `isMain` has always been here, but it wasn't - it was added by #5950 which didn't work out. 

It would be better (for the sake of a cleaner history) to start clean with a new solution.

Using `isMain` introduces different behaviour for the node called from the command line and whenever things are included as a module. For example, why behavior of node invoked as the "install" or "postinstall" script from the "package.json" should be different when the module was included? 

When troubleshooting a module load, a simple `node -e 'require("module");' may behave differently as if `require("module");` got invoked from some other module. Some modules that are used standalone can be also included in the test modules, in this case the behaviour here will be different. This can be even more frustrating than the symbolic link behaviour contested in #3402.